tests/benchmarks/test_xgboost.py

```diff
@@ -0,0 +1,237 @@
+    ddf["airport_fee"] = ddf["airport_fee"].replace("None", 0).astype(float).fillna(0)
+
+    #############
+    # Filter rows
+    #############
+    ddf = ddf.dropna(how="any")
+    # Remove outliers
+    lower_bound = 0
+    Q3 = ddf["trip_time"].quantile(0.75)
+    upper_bound = Q3 + (1.5 * (Q3 - lower_bound))
+    ddf = ddf.loc[(ddf["trip_time"] >= lower_bound) & (ddf["trip_time"] <= upper_bound)]
+
+    ###################
+    # Join with domains
+    ###################
+    ddf = dd.merge(
+        ddf,
+        taxi_zone_lookup,
+        how="inner",
+        left_on="PULocationID",
+        right_on="LocationID",
+    )
+    ddf = ddf.rename(columns={"Borough": "PUBorough", "Superborough": "PUSuperborough"})
+    ddf = ddf.drop(columns="LocationID")
+
+    ddf = dd.merge(
+        ddf,
+        taxi_zone_lookup,
+        how="inner",
+        left_on="DOLocationID",
+        right_on="LocationID",
+    )
+    ddf = ddf.rename(columns={"Borough": "DOBorough", "Superborough": "DOSuperborough"})
+    ddf = ddf.drop(columns="LocationID")
+
+    ddf["PUSuperborough_DOSuperborough"] = ddf.PUSuperborough.str.cat(
+        ddf.DOSuperborough, sep="-"
+    )
+    ddf = ddf.drop(columns=["PUSuperborough", "DOSuperborough"])
+
+    ############
+    # Categorize
+    ############
+    categories = [
+        "hvfhs_license_num",
+        "PULocationID",
+        "DOLocationID",
+        "wav_request_flag",
+        "accessible_vehicle",
+        "pickup_month",
+        "pickup_dow",
+        "pickup_hour",
+        "PUBorough",
+        "DOBorough",
+        "PUSuperborough_DOSuperborough",
+    ]
+
+    # TODO https://github.com/dask/dask/issues/9847
+    ddf = ddf.astype(dict.fromkeys(categories, "category"))
+    ddf = ddf.persist()
+    # This blocks until the whole workload so far has been persisted
+    ddf = ddf.categorize(categories)
+
+    ########
+    # Output
+    ########
+    ddf = ddf.persist().repartition(partition_size="100MB")
+    # At this point we would normally finish with to_parquet()
+    wait(ddf, small_client, timeout=600)
+
+
+@run_up_to_nthreads("small_cluster", 200, reason="fixed size dataset")
+def test_optuna_hpo(small_client):
+    # Needs either coiled-runtime >0.2.1 or explicitly adding the package to the A/B
+    # test environment
+    optuna = pytest.importorskip("optuna")
+
+    #############################
+    # Dataset load and preprocess
+    #############################
+    # Load feature table generated by previous test
+    ddf = dd.read_parquet(
+        "s3://coiled-runtime-ci/xgboost/nyc-uber-lyft-preprocessed.parquet",
+    )
+
+    # Full size dataset on 40x r6i.large
+    ddf = downscale_dataframe(ddf, "580 GiB")
+
+    # Under the hood, XGBoost converts floats to `float32`.
+    # Let's do it only once here.
+    float_cols = ddf.select_dtypes(include="float").columns.tolist()
+    ddf = ddf.astype({c: np.float32 for c in float_cols})
```

Comment:
I think the data was already saved in float32, do we need explicitly np.float32? Not sure I follow this part.

Reply:
airport_fee is float64

Reply:
Oh, I assumed it was float64 since in the code used to write this data it was supposed to convert it. I'm looking here https://github.com/coiled/coiled-from-prefect/blob/0a19e2daed1afe3fc43501f1ad4c0be470f9c232/flows/coiled_flow.py#L69-L80

I guess something went wrong, in that case, it makes sense to convert it again. That being said I was not able to find in their docs that this is what happens.